File: Coding Exercises/Python/Day46/is_armstrong_number.py
def is_armstrong_number(num):
    original_number, result, n = num, 0, 0

    while original_number != 0:
        original_number //= 10
        n += 1

    original_number = num

    while original_number != 0:
        remainder = original_number % 10
        result += remainder ** n
        original_number //= 10

    return result == num

File: Coding Exercises/TypeScript/Day46/isArmstrongNumber.ts
function isArmstrongNumber(num: number): boolean {
  let originalNumber: number = num;
  let result: number = 0;
  let n: number = 0;

  while (originalNumber !== 0) {
      originalNumber = Math.floor(originalNumber / 10);
      n++;
  }

  originalNumber = num;

  while (originalNumber !== 0) {
      const remainder: number = originalNumber % 10;
      result += Math.pow(remainder, n);
      originalNumber = Math.floor(originalNumber / 10);
  }

  return result === num;
}
